ui/imports/shared/popups/ImportCommunityPopup.qml

```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
-
-            const details = root.store.getCommunityDetails(key)
-            if (!!details) // the above can return `null` in which case we continue loading
-                loading = false
-            return details
-        }
+        property bool communityFound: !!d.communityDetails && !!d.communityDetails.name
+        property var communityDetails: null
+
+        property var requestedCommunityDetails: null
 
         readonly property string inputErrorMessage: isInputValid ? "" : qsTr("Invalid key")
         readonly property string errorMessage: importErrorMessage || inputErrorMessage
         readonly property string inputKey: keyInput.text.trim()
-        readonly property bool isPrivateKey: (Utils.isPrivateKey(inputKey))
-        readonly property bool isPublicKey: (publicKey !== "")
-        readonly property string privateKey: inputKey
         readonly property string publicKey: {
-            if (!Utils.isStatusDeepLink(inputKey)) {
-                const key = Utils.dropCommunityLinkPrefix(inputKey)
-                if (!Utils.isCommunityPublicKey(key))
-                    return ""
-                if (!Utils.isCompressedPubKey(key))
-                    return key
-                return Utils.changeCommunityKeyCompression(key)
-            } else {
-                return Utils.getCommunityDataFromSharedLink(inputKey).communityId;
+            if (Utils.isStatusDeepLink(inputKey)) {
+                const linkData = Utils.getCommunityDataFromSharedLink(inputKey)
+                return !linkData ? "" : linkData.communityId
             }
+            if (!Utils.isCommunityPublicKey(inputKey))
+                return ""
+            if (!Utils.isCompressedPubKey(inputKey))
+                return inputKey
+            return Utils.changeCommunityKeyCompression(inputKey)
+        }
+        readonly property bool isInputValid: publicKey !== ""
+
+        property bool communityInfoRequested: false
+
+        function updateCommunityDetails(requestIfNotFound) {
+            if (!isInputValid) {
+                d.communityInfoRequested = false
+                d.communityDetails = null
+                return
+            }
+
+            const details = root.store.getCommunityDetails(publicKey)
+
+            if (!!details) {
+                d.communityInfoRequested = false
+                d.communityDetails = details
+                return
+            }
+
+            if (requestIfNotFound) {
+                root.store.requestCommunityInfo(publicKey, false)
+                d.communityInfoRequested = true
+                d.communityDetails = null
+            }
+        }
+
+        onPublicKeyChanged: {
+            // call later to make sure all proeprties used by `updateCommunityDetails` are udpated
+            Qt.callLater(() => { d.updateCommunityDetails(true) })
         }
-        readonly property bool isInputValid: isPrivateKey || isPublicKey
     }
 
     Timer {
```

Comment:
Uhm, remove the `Timer` when it's not used?

Reply:
oh, totally missed to remove!